packages/lesswrong/components/dialogues/CalendlyIFrame.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import { registerComponent } from '../../lib/vulcan-lib'
+import { calendlyPreviewStyles } from '../../themes/stylePiping'
+
+const styles = (theme: ThemeType) => ({
+  ...calendlyPreviewStyles(theme),
+  calendlyEmbed: {
+    ...calendlyPreviewStyles(theme)["& div.calendly-preview"]
+  }
+})
+
+export const validatedCalendlyUrl = (url: string) => {
+  if (url === "") return {valid: true, url: ""}
+  let calendlyUrl = null
+  try {
+    calendlyUrl = new URL(url.slice(0,4) === "http" ? url : `https://${url}`)
+    const valid = calendlyUrl.hostname.match(/^(www\.)?calendly.com$/) != null
```

Comment:
`!=` double-eq?

Reply:
also I am scared of regex matching for validation, but I guess it is better that it's going through the `URL` constructor first.

Reply:
I'm doing `!=` for null/undefined type stuff, but I guess it should be actual null rather than undefined

Reply:
> An [Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) whose contents depend on the presence or absence of the global (g) flag, or [null](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null) if no matches are found.

Reply:
Also, you probably need to also make a `createValidate` callback hook for `dialogueMatchPreferences` that performs the same validation server-side, since otherwise someone can pass in some totally different URL by hitting the API directly.